components/builder-originsrv/src/migrations/origin_packages.rs

```diff
@@ -42,44 +42,94 @@ pub fn migrate(migrator: &mut Migrator) -> Result<()> {
+                        op_checksum text,
+                        op_manifest text,
+                        op_config text,
+                        op_target text,
+                        op_deps text,
+                        op_tdeps text,
+                        op_exposes text
+                    ) RETURNS TABLE (
+                        id bigint,
+                        name text,
+                        origin_id bigint,
+                        owner_id bigint,
+                        ident text,
+                        checksum text,
+                        manifest text,
+                        config text,
+                        target text,
+                        deps text,
+                        tdeps text,
+                        exposes text,
+                        channels text[]
+                    ) AS $$
+                        DECLARE
+                            channel_id bigint;
+                            package_id bigint;
+                        BEGIN
+                            INSERT INTO origin_packages (origin_id, owner_id, name, ident, checksum, manifest, config, target, deps, tdeps, exposes)
+                            VALUES (op_origin_id, op_owner_id, op_name, op_ident, op_checksum, op_manifest, op_config, op_target, op_deps, op_tdeps, op_exposes)
+                            RETURNING origin_packages.id INTO package_id;
+
+                            SELECT oc.id FROM origin_channels oc WHERE oc.origin_id = op_origin_id AND oc.name = 'unstable'
+                              INTO channel_id;
 
-                         SELECT id FROM origin_channels WHERE origin_id = op_origin_id AND name = 'unstable' INTO channel_id;
-                         PERFORM promote_origin_package_v1(channel_id, inserted_package.id);
+                            PERFORM promote_origin_package_v1(channel_id, package_id);
 
-                         RETURN NEXT inserted_package;
-                         RETURN;
-                     END
-                 $$ LANGUAGE plpgsql VOLATILE"#)?;
+                            RETURN QUERY
+                                SELECT p.id, p.name, p.origin_id, p.owner_id, p.ident, p.checksum, p.manifest,
+                                       p.config, p.target, p.deps, p.tdeps, p.exposes,
+                                    (SELECT ARRAY(SELECT c.name
+                                                    FROM origin_channels c
+                                                   WHERE c.id IN (SELECT cp.channel_id
+                                                                    FROM origin_channel_packages cp
+                                                                   WHERE cp.package_id = p.id
+                                                                ORDER BY cp.created_at DESC))) AS channels
+                                  FROM origin_packages p
+                                 WHERE p.id = package_id;
+                            RETURN;
```

Comment:
Yeah, this looks like it's performing essentially the same query as the earlier function, but without the channel restriction. This really makes me favor a view for this logic now; it encapsulates all the join logic, but you can modify the selection criteria when you call it. As a bonus, you can also refer to the view as a return type, so you don't need to painstakingly list all the columns each time. With a concrete type, I think you can also indicate that you're only going to be returning a single row instead of a table (that you have to remember only ever has at most one row), but that's not 100% necessary.